network/network.go

```diff
@@ -193,12 +193,12 @@ type InterfaceInfo struct {
 	// Address contains an optional static IP address to configure for
 	// this network interface. The subnet mask to set will be inferred
 	// from the CIDR value.
-	Address corenetwork.Address
+	Address corenetwork.ProviderAddress
```

Comment:
This all needs relocating to `core/network` at some point, this patch is large enough as is.

Reply:
Please add this as a comment TODO